modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableViewSkinBase.java

```diff
@@ -573,48 +579,96 @@ void horizontalScroll() {
         tableHeaderRow.updateScrollX();
     }
 
-    void onFocusPreviousCell() {
-        TableFocusModel<M,?> fm = getFocusModel();
+    /**
+     * Called when the focus is set on previous cell in order to scroll to it to make it visible. This usually happens
+     * when the focus is set on the cell above the current focused cell.
+     *
```

Comment:
I said "on the cell above the current focused cell". I really don't see how you managed to see it on the left? 
We are dealing with the TableView that lays its cells into rows. There is no orientation that can be chosen on the TableView regarding the VirtualFlow. The VirtualFlow orientation will always be vertical meaning the Virtualization will be on rows.

I believe above is a pretty straight forward when you consider the TableView layout :
cell11 cell12 cell13
cell21 cell22 cell23
cell31 cell32 cell33

If I'm considering cell22, above is refering to cell12. I don't see how it could be mistaken by cell21. Or if it is, I'm lacking words to describe it more accurately.

Reply:
Maybe the confusing part is "This *usually* happens when", it implies that there are cases where the cell is not the one above.

Reply:
In fact, I'm not really sure about the documentation. The TableView has a Behavior that will execute the selection and the focus. All these methods are called like that :
```
behavior.setOnFocusPreviousRow(() -> onFocusPreviousCell());
        behavior.setOnFocusNextRow(() -> onFocusNextCell());
        behavior.setOnMoveToFirstCell(() -> onMoveToFirstCell());
        behavior.setOnMoveToLastCell(() -> onMoveToLastCell());
        behavior.setOnScrollPageDown(isFocusDriven -> onScrollPageDown(isFocusDriven));
        behavior.setOnScrollPageUp(isFocusDriven -> onScrollPageUp(isFocusDriven));
        behavior.setOnSelectPreviousRow(() -> onSelectPreviousCell());
        behavior.setOnSelectNextRow(() -> onSelectNextCell());
        behavior.setOnSelectLeftCell(() -> onSelectLeftCell());
        behavior.setOnSelectRightCell(() -> onSelectRightCell());
```

So basically, the onFocusNextCell and onFocusPreviousCell are doing the exact same thing. They scroll to make the cell visible. I used the word "usually" because I do not like the method naming. I believe there should only be one method called "scrollToFocusedCell". And then the description would be easier. 
Right now we have a method "onFocusNextCell" that is called when the focus is changed, but anyone could call it to simply scroll to the focused index..

Reply:
So I suggest you discuss this in a comment here or in the mailing list. Maybe `onFocusPreviousCell` and `onFocusNextCell` can be combined to one. `onSelectPreviousCell` and `onSelectNextCell` are a similar case.

Reply:
If we decide to merge them as one, we loose the information whether the method was called because the TableViewBehavior has selected the previous or the next cell. 
In my personal case, I don't care since I simply scroll to make the cell visible. So does the TableView. Maybe people are interested in that information?

Reply:
Maybe, this is something to discuss with everyone. In any case, I think that "previous" and "next" are not the best descriptors. "up"/"down" or "above"/"below" are more concrete in my opinion.

Reply:
They are called by the behavior when "setOnSelectPreviousRow" is called. "Previous" row makes sense. But the "previous cell" does not really indeed. Especially that the behavior methods are mapped to the "up" and "down" keyboard buttons. 
What's the best way to collect opinions? The mailing list or here?

Reply:
Probably the mailing list.

Reply:
wrote up [some thoughts](https://github.com/kleopatra/swingempire-fx/wiki/TableSelectionModel) on naming. The executive version:

- let the naming be driven by the naming of the corresponding selection/focusModel methods
- use consistent names across skin/behavior
- be complete, that is let behavior/skin provide api hooks for a action/hook methods for all model navigational methods that are used by them

This would be partly out off scope of the core bug, as it would require changes in behavior as well (change xxPreviousRow to xxAboveCell and add additional methods for taking callback hooks). Could mitigate by defering the completeness to future and align the skin names to the corresponding behavior names - they are acceptable, IMO.

Reply:
Thanks for the feedback, few thoughts:
- The TableView has two selection Mode : Cell based or row based. Maybe the confusion is the TableViewBehavior comes from that. `SelectLeftCell ` is acceptable because it will only be called when the TableView is in cell based selection. 
However, selecting the "previous" is complicate to name. If you are row based, you'll select the previous row (and that make sense because in row based selection, you only have the next row and the previous row). But if you are cell based, you'll select the cell above. So the naming of the method that will be called when the key "UP" is pressed is not trivial.

Another thing I wanted to share is also the fact that there is no "onFocusLeftCell" in the TableSkin. Right now we have in the Behavior :
```
onSelectNextRow
onSelectPreviousRow
onSelectRightCell
onSelectLeftCell
```
But we only have :
```
onFocusPreviousRow
onFocusNextRow
```
And if we look at the TableViewBehaviorBase :
```
protected void focusLeftCell() {
        TableSelectionModel sm = getSelectionModel();
        if (sm == null) return;
        TableFocusModel fm = getFocusModel();
        if (fm == null) return;

        fm.focusLeftCell();

        if (onFocusPreviousRow != null) onFocusPreviousRow.run();
    }
```
So we are calling `onFocusPreviousRow ` when we are focusing the left cell. So I don't know what the final decision will be, but if we want to tackle these naming issue, we may want to address what I've stated above also.

Reply:
@kleopatra After trying to help with the documentation I'm realizing fast that the API is not standard. Maybe you could help with creating a normal API with the people who are familiar with it.